### PR TITLE
chore: Update Rust dependencies

### PR DESCRIPTION
**Pull request by Techassi**
# Description

This bumps various Rust dependencies which are not yet handled by #811

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

### CODE REVIEW

**Review by nightkr — APPROVED** [review]

**Comment by Techassi** [comment]
Forgot to update the CRD, re-requesting review.

File: rust/crd/src/affinity.rs [hunk 1]
```diff
@@ -95,6 +95,11 @@ mod tests {
                             namespace_selector: None,
                             namespaces: None,
                             topology_key: "kubernetes.io/hostname".to_string(),
+                            // NOTE (@Techassi): Both these fields were added in
+                            // Kubernetes 1.30, and cannot be used by use for
+                            // now.
```
Comment (sbernauer) [inline rust/crd/src/affinity.rs:100]:
```suggestion
                            // Kubernetes 1.30, and cannot be used for now.
```

**Review by sbernauer — DISMISSED** [review]
LGTM otherwise

File: rust/crd/src/affinity.rs (hunk 1 above)
Comment (Techassi) [reply rust/crd/src/affinity.rs:100]: 80 characters wide tho :eyes:

**Review by sbernauer — APPROVED** [review]

**Pull request merged**